counsel.el

```diff
@@ -3127,6 +3127,26 @@ otherwise continue prompting for tags."
 ;;;###autoload
 (defalias 'counsel-org-goto #'counsel-outline)
 
+(defcustom counsel-org-goto-all-outline-path-prefix nil
+  "When the value is 'file, include the file name (without
+directory) into the outline path candidate.
+
+When 'full-file-path, include the full file name.
+
+When 'buffer-name, include the buffer name."
+  :type '(choice (const :tag "None" nil)
+	         (const :tag "File name" file)
+	         (const :tag "Full file name" full-file-path)
+	         (const :tag "Buffer name" buffer-name)))
+
+(defun counsel-org-goto-all--outline-path-prefix ()
+  (cl-case counsel-org-goto-all-outline-path-prefix
+    (file (file-name-nondirectory (buffer-file-name)))
+    (full-file-path (buffer-file-name))
```

Comment:
Tip: the buffer-local variable `buffer-file-name` holds the same value as that returned by calling the function `buffer-file-name` with no arguments.